modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/FunctionObjectEntryManager.java

```diff
@@ -61,6 +65,10 @@ public ObjectEntry addObjectEntry(
 			String scopeKey)
 		throws Exception {
 
+		checkPortletResourcePermission(
```

Comment:
to avoid this code duplication we need to create another Base for non-default object entry managers, 

otherwise this check permission will be applied for the DefaultObjectEntryManagerImpl and it's not the expected,

we could use the [template method approach](https://refactoring.guru/design-patterns/template-method):

```java
BaseObjectEntryManager {
 
 public ObjectEntry addObjectEntry(
	    DTOConverterContext dtoConverterContext,
	    ObjectDefinition objectDefinition, ObjectEntry objectEntry,
	    String scopeKey)
    throws Exception {
               
         checkPermission(...)

	 executeAddObjectEntry(...)
 }
}
```
so, for DefaultObjectEntryManagerImpl:

```java
...

public void checkPermission(){
  // do nothing because this layer is not suppose to check permissions
}

public void executeAddObjectEntry(){
  // do _objectEntryService.addObjectEntry(... 
}
...
```

non-default:

```java
...

public void checkPermission(){
  // check permission
}

public void executeAddObjectEntry(){
  // do _portalCatapult.launch( ... or Salesforce ...
}
...
```

Reply:
@guilhermedcamacho do you see any other apporach that can be simpler than this one?